### ISSUE TITLE
Cortex has widespread s3 client errors when run on AWS/us-east-1

### ISSUE THREAD

**Issue by nickbp** [opstrace/opstrace#883]
**Describe the bug**
These cortex components (in the `cortex` namespace) fail to run in an AWS opstrace instance that's been deployed to `us-east-1`. It looks like any cortex component that touches S3 has problems. Going by prior issues with us-east-1 in #841, it's likely that S3 in us-east-1 is behaving in a way that's incompatible with all other regions. Oddly enough, loki pods are not showing the same problems, so there may be different S3 clients being used between cortex and loki.

For the record, I think it's reasonable to just say that we don't support this region, and to recommend that users go with other regions, since who knows what other gotchas there are here (we've already hit one via the installer as it is). I have specifically heard this region called "the most duct-taped" among AWS regions.

- alertmanager:
```
level=error ts=2021-06-15T01:11:46.042553268Z caller=cortex.go:447 msg="module failed" module=alertmanager err="invalid service state: Failed, expected: Running, failure: failed to list users with alertmanager configuration: Get \"https://scaletest-cortex-config.s3.dualstack.us-east-1.amazonaws.com/?delimiter=%2F&encoding-type=url&fetch-owner=true&list-type=2&prefix=alerts%2F\": 301 response missing Location header"
```
- querier:
```
level=error ts=2021-06-15T01:10:52.659070225Z caller=blocks_finder_bucket_scan.go:157 msg="unable to run the initial blocks scan" err="Get \"https://scaletest-cortex.s3.dualstack.us-east-1.amazonaws.com/?delimiter=%2F&encoding-type=url&fetch-owner=true&list-type=2&prefix=\": 301 response missing Location header"
level=error ts=2021-06-15T01:10:52.659178127Z caller=cortex.go:447 msg="module failed" module=store-queryable err="invalid service state: Failed, expected: Running, failure: unable to start blocks storage queryable subservices: not healthy, 0 terminated, 1 failed: [Get \"https://scaletest-cortex.s3.dualstack.us-east-1.amazonaws.com/?delimiter=%2F&encoding-type=url&fetch-owner=true&list-type=2&prefix=\": 301 response missing Location header]"
level=info ts=2021-06-15T01:10:52.659264283Z caller=module_service.go:91 msg="module stopped" module=distributor-service
level=error ts=2021-06-15T01:10:52.659263391Z caller=cortex.go:447 msg="module failed" module=querier err="failed to start querier, because it depends on module store-queryable, which has failed: invalid service state: Failed, expected: Running, failure: invalid service state: Failed, expected: Running, failure: unable to start blocks storage queryable subservices: not healthy, 0 terminated, 1 failed: [Get \"https://scaletest-cortex.s3.dualstack.us-east-1.amazonaws.com/?delimiter=%2F&encoding-type=url&fetch-owner=true&list-type=2&prefix=\": 301 response missing Location header]"
```
- ruler:
```
level=error ts=2021-06-15T01:11:09.011029549Z caller=blocks_finder_bucket_scan.go:157 msg="unable to run the initial blocks scan" err="Get \"https://scaletest-cortex.s3.dualstack.us-east-1.amazonaws.com/?delimiter=%2F&encoding-type=url&fetch-owner=true&list-type=2&prefix=\": 301 response missing Location header"
level=error ts=2021-06-15T01:11:09.011143277Z caller=cortex.go:447 msg="module failed" module=store-queryable err="invalid service state: Failed, expected: Running, failure: unable to start blocks storage queryable subservices: not healthy, 0 terminated, 1 failed: [Get \"https://scaletest-cortex.s3.dualstack.us-east-1.amazonaws.com/?delimiter=%2F&encoding-type=url&fetch-owner=true&list-type=2&prefix=\": 301 response missing Location header]"
level=error ts=2021-06-15T01:11:09.011206627Z caller=cortex.go:447 msg="module failed" module=ruler err="failed to start ruler, because it depends on module store-queryable, which has failed: invalid service state: Failed, expected: Running, failure: invalid service state: Failed, expected: Running, failure: unable to start blocks storage queryable subservices: not healthy, 0 terminated, 1 failed: [Get \"https://scaletest-cortex.s3.dualstack.us-east-1.amazonaws.com/?delimiter=%2F&encoding-type=url&fetch-owner=true&list-type=2&prefix=\": 301 response missing Location header]"
level=info ts=2021-06-15T01:11:09.011217567Z caller=module_service.go:91 msg="module stopped" module=distributor-service
```
- store-gateway:
```
level=info ts=2021-06-15T01:08:26.262698189Z caller=bucket_stores.go:142 msg="synchronizing TSDB blocks for all users"
level=warn ts=2021-06-15T01:08:42.276518982Z caller=bucket_stores.go:147 msg="failed to synchronize TSDB blocks" err="Get \"https://scaletest-cortex.s3.dualstack.us-east-1.amazonaws.com/?delimiter=%2F&encoding-type=url&fetch-owner=true&list-type=2&prefix=\": 301 response missing Location header"
level=info ts=2021-06-15T01:08:42.276604798Z caller=basic_lifecycler.go:197 msg="ring lifecycler is shutting down" ring=store-gateway
level=info ts=2021-06-15T01:08:42.276807943Z caller=basic_lifecycler.go:363 msg="unregistering instance from ring" ring=store-gateway
level=info ts=2021-06-15T01:08:42.276866821Z caller=basic_lifecycler.go:234 msg="instance removed from the ring" ring=store-gateway
level=error ts=2021-06-15T01:08:42.276933902Z caller=cortex.go:447 msg="module failed" module=store-gateway err="invalid service state: Failed, expected: Running, failure: initial blocks synchronization: Get \"https://scaletest-cortex.s3.dualstack.us-east-1.amazonaws.com/?delimiter=%2F&encoding-type=url&fetch-owner=true&list-type=2&prefix=\": 301 response missing Location header"
```

**To Reproduce**
Deploy an opstrace cluster on AWS/`us-east-1` specifically, then watch the pods in the `cortex` namespace for errors/restarts.

**Comment by nickbp** [comment]
Actually the problem might just be PEBKAC:
```
curl -SLIXGET "https://scaletest-cortex.s3.dualstack.us-east-1.amazonaws.com/?delimiter=%2F&encoding-type=url&fetch-owner=true&list-type=2&prefix="
HTTP/1.1 301 Moved Permanently
x-amz-bucket-region: us-west-1
```

I had deployed to us-west-1 earlier, but they're apparently having a node shortage so then I tried us-east-1. I hadn't manually deleted the S3 buckets afterwards. Could be something for the installer to catch?

So things might actually work fine if you don't have s3 buckets left over in other regions like I did.
